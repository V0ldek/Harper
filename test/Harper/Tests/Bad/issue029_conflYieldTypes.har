generate :: Integer -> Iterator Integer;
generate n = {
  var (i :: Integer) = 0;
  while i < n {
    yield i;
    i += 1;
  }
  yield (i, n);
};

main :: ();
main = ();