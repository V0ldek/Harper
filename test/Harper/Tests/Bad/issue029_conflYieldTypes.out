Error: cannot unify conflicting yield types of a function: `(Integer, Integer)`, `Integer`.
During evaluation of:
{
  var (i :: Integer)= 0 ;
  while i < n {
    yield i ;
    i += 1 ;
  }
  yield (i, n);
}
Located at line 2 column 14

Execution terminated with an error: type error.